### PR TITLE
New version of service o365_mu with resource mailboxes included

### PR DESCRIPTION
**Pull request by stavamichal**
- add new objects "resource-mail boxes" to the service (lots of new
   attributes and whole new logic)
 - logic was add to gen script o365_mu (to support new objects in the
   new file with these data)
 - and also was add to send script part (to be able to process new data)

### CODE REVIEW

File: send/o365_mu_process.pl [hunk 1]
```diff
@@ -446,6 +598,91 @@ sub readDataAboutGroups {
...
+#Sub to read data about resource-mails from file and convert it to perl hash
+sub readDataAboutResourceMails {
+	my $pathToFile = shift;
+
+	my $resourceMailsStruc = {};
+	open FILE, $pathToFile or die "Could not open file with resource-mails data from perun $pathToFile: $!\n";
+
+	while(my $line = <FILE>) {
+		chomp( $line );
+    my @parts = split /\t/, $line;
```
Comment (licehammer) [inline send/o365_mu_process.pl]: Indentation.

File: send/o365_mu_process.pl [hunk 2]
```diff
@@ -229,11 +305,17 @@
 	print FILE_GROUPS_CACHE $groupRecord . "\n";
 }
 
+foreach my $resourceMailRecord (@allResourceMails) {
```
Comment (licehammer) [inline send/o365_mu_process.pl]: Variable `@allResourceMails` is guarded by `$RESOURCE_MAILS_LOCK`. I'm aware that in this spot is the only one thread active. Therefore it's no need for the lock. But for the sake of future refactoring or changes in this script, I would put a lock here anyway or at least the comment explaining why the lock is not necessary.

File: send/o365_mu_process.pl [hunk 3]
```diff
@@ -446,6 +598,91 @@ sub readDataAboutGroups {
...
+		$resourceMailsStruc->{$resourceMailName}->{$RES_WORKING_HOURS_END_TIME_TEXT} = $resourceWorkingHoursEndTime;
+		$resourceMailsStruc->{$resourceMailName}->{$RES_ALLOW_RECURRING_MEETINGS_TEXT} = $resourceAllowRecurringMeetings;
+		$resourceMailsStruc->{$resourceMailName}->{$RES_ADD_ADDITIONAL_RESPONSE_TEXT} = $resourceAddAdditionalResponse;
+		$resourceMailsStruc->{$resourceMailName}->{$RES_DELEGATES_TEXT} = \@resourceDelegates;
+		$resourceMailsStruc->{$resourceMailName}->{$RES_BOOK_IN_POLICY_TEXT} = \@resourceBookInPolicy;
+		$resourceMailsStruc->{$resourceMailName}->{$RES_REQUEST_IN_POLICY_TEXT} = \@resourceRequestInPolicy;
+		$resourceMailsStruc->{$resourceMailName}->{$RES_REQUEST_OUT_OF_POLICY_TEXT} = \@resourceRequestOutOfPolicy;
+		$resourceMailsStruc->{$resourceMailName}->{$PLAIN_TEXT_OBJECT_TEXT} = $line;
+	}
+  close FILE or die "Could not close file $pathToFile: $!\n";
```
Comment (licehammer) [inline send/o365_mu_process.pl]: Indent.

**Comment by stavamichal** [comment]
All should be fixed now. For these missing locks I added a comment.

**Comment by stavamichal** [comment]
We need to add also part with removing not existing resource before merging this pull request.

**Comment by stavamichal** [comment]
New changes were made, review need to be done again please.

File: send/o365_mu_process.pl [hunk 4]
```diff
@@ -274,20 +361,41 @@ sub processTasks {
...
+		cond_broadcast($BARRIER_LOCK);
+	} else {
+		cond_wait($BARRIER_LOCK);
+	}
+}
+
+#Sub to wait for every running thread and synchronize them
+sub waitForSynchronize {
+	#send special job to queue to signal that all threads should wait for others to synchronize
+  for(1 .. $THREAD_COUNT) {
```
Comment (licehammer) [inline send/o365_mu_process.pl]: Wrong indentation.

File: send/o365_mu_process.pl [hunk 5]
```diff
@@ -274,20 +361,41 @@ sub processTasks {
...
+  }
+}
+
+#Sub to wait for every running thread and send terminate operation for all of them
 sub waitForThreads {
 	#send special job to queue to signal the thread to terminate
-	for(1 .. $THREAD_COUNT) {
-		$jobQueue->enqueue( { $OPERATION => $OPERATION_END } );
-	}
+  for(1 .. $THREAD_COUNT) {
```
Comment (licehammer) [inline send/o365_mu_process.pl]: Wrong indentation.

**Comment by stavamichal** [comment]
Indentation fixed.

**Pull request merged**